pkg/jobs/registry.go

```diff
@@ -1993,28 +2002,66 @@ func (r *Registry) WaitForRegistryShutdown(ctx context.Context) {
 }
 
 // DrainRequested informs the job system that this node is being drained.
-// Returns a function that, when invoked, will initiate drain process
-// by requesting all currently running jobs, as well as various job registry
-// processes terminate.
+// Waits up to maxWait time for any callers who are monitoring for registry
+// drain to complete their drain logic.
 // WaitForRegistryShutdown can then be used to wait for those tasks to complete.
-func (r *Registry) DrainRequested() func() {
+func (r *Registry) DrainRequested(ctx context.Context, maxWait time.Duration) {
 	r.mu.Lock()
 	alreadyDraining := r.mu.draining
+	numWait := r.mu.numDrainWait
 	r.mu.draining = true
 	r.mu.Unlock()
 
 	if alreadyDraining {
-		return func() {}
+		return
```

Comment:
This is existing code and fine as-is, but I mildly wonder whether we should actually return some assertion failure error here rather than just early exiting since we expect only one holder of the registry to call this.

Reply:
I wonder if drain logic is idempotent -- like may drain RPC replay or something could cause this to be called
multiple times.  I think it's fine -- we are still performing a 1 way transition.
Perhaps drain logic has strong guarantees, but maybe we shouldn't rely on that.

Reply:
yes, the drain logic can be called multiple times and is meant to be idempotent.